packages/devkit/src/executors/parse-target-string.ts

```diff
@@ -1,10 +1,8 @@
 import type { Target } from 'nx/src/command-line/run/run';
 import type { ProjectGraph } from 'nx/src/config/project-graph';
-// eslint-disable-next-line @typescript-eslint/no-restricted-imports
-import { splitTarget } from 'nx/src/utils/split-target';
 import { requireNx } from '../../nx';
 
-const { readCachedProjectGraph } = requireNx();
+const { readCachedProjectGraph, splitTarget } = requireNx();
```

Comment:
`splitTarget` here would be null if the installed nx version is prior to whenever this PR lands, so its usage needs to be null checked. I'd imagine this would look something like:

```javascript
const splitTarget = requireNx().splitTarget ?? require('nx/src/utils/split-target');
```

The require call wouldn't fire if splitTarget is present, and then we can drop it after Nx 18 or when we are confident that all of devkit's supported Nx versions should have it available.

Reply:
Something like this will be needed for each of the APIs moved to devkit internals